File: app/src/main/java/com/asfoundation/wallet/repository/CurrencyConversionService.java
package com.asfoundation.wallet.repository;

import com.asfoundation.wallet.service.TokenRateService;
import com.asfoundation.wallet.service.LocalCurrencyConversionService;
import com.asfoundation.wallet.ui.iab.FiatValue;
import io.reactivex.Single;

/**
 * Created by franciscocalado on 24/07/2018.
 */

public class CurrencyConversionService {

  private final TokenRateService tokenRateService;
  private final LocalCurrencyConversionService localCurrencyConversionService;

  public CurrencyConversionService(TokenRateService tokenRateService,
      LocalCurrencyConversionService localCurrencyConversionService) {
    this.tokenRateService = tokenRateService;
    this.localCurrencyConversionService = localCurrencyConversionService;
  }

  public Single<FiatValue> getTokenValue(String currency) {
    return tokenRateService.getAppcRate(currency);
  }

  public Single<FiatValue> getLocalFiatAmount(String appcValue) {
    return localCurrencyConversionService.getAppcToLocalFiat(appcValue).firstOrError();
  }
}

File: app/src/main/java/com/asfoundation/wallet/repository/CurrencyConversionService.kt
package com.asfoundation.wallet.repository

import com.asfoundation.wallet.service.TokenRateService
import com.appcoins.wallet.feature.changecurrency.data.currencies.LocalCurrencyConversionService
import com.appcoins.wallet.feature.changecurrency.data.currencies.FiatValue
import io.reactivex.Single
import javax.inject.Inject

class CurrencyConversionService @Inject constructor(private val tokenRateService: TokenRateService,
                                private val localCurrencyConversionService: LocalCurrencyConversionService) {

  fun getTokenValue(currency: String?): Single<FiatValue> {
    return tokenRateService.getAppcRate(currency)
  }

  fun getLocalFiatAmount(appcValue: String): Single<FiatValue> {
    return localCurrencyConversionService.getAppcToLocalFiat(appcValue, 18)
  }

  fun getLocalFiatAmount(value: String, currency: String): Single<FiatValue> {
    return localCurrencyConversionService.getFiatToLocalFiat(currency, value, 2)
  }

  fun getFiatToAppcAmount(value: String, currency: String): Single<FiatValue> {
    return localCurrencyConversionService.getFiatToAppc(currency, value, 18)
  }
}